Zend/zend_compile.c

```diff
@@ -1456,8 +1463,14 @@ static zend_bool zend_try_ct_eval_class_const(zval *zv, zend_string *class_name,
 		return 0;
 	}
 
+	if (!cc || !zend_verify_const_access(cc, CG(active_class_entry))) {
+		return 0;
+	}
+
+	c = &cc->value;
+
 	/* Substitute case-sensitive (or lowercase) persistent class constants */
-	if (c && Z_TYPE_P(c) < IS_OBJECT) {
+	if (Z_TYPE_P(c) < IS_OBJECT) {
 		ZVAL_DUP(zv, c);
```

Comment:
I also think ZVAL_COPY() should be enough, but lets change this in a
separate patch.

On Mon, Dec 7, 2015 at 7:27 PM, Xinchen Hui <notifications@github.com>
wrote:

> In Zend/zend_compile.c
> <https://github.com/php/php-src/pull/1662#discussion_r46841212>:
>
> >  	/* Substitute case-sensitive (or lowercase) persistent class constants */
> > -	if (c && Z_TYPE_P(c) < IS_OBJECT) {
> > +	if (Z_TYPE_P(c) < IS_OBJECT) {
> >  		ZVAL_DUP(zv, c);
>
> I know it's not introduced by this patch, but why ZVAL_DUP here? I think
> ZVAL_COPY here should be enough
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1662/files#r46841212>.
>